src/test/java/com/cloudant/tests/CloudantClientTests.java

```diff
@@ -266,67 +276,40 @@ public void connectionTimeout() throws Throwable {
      * server waits 0.5 s before continuing to send data down the stream so the timeout should be
      * triggered.
      */
-    @Test(expected=SocketTimeoutException.class)
+    @Test(expected = SocketTimeoutException.class)
     public void readTimeout() throws Throwable {
 
-        final CountDownLatch startupLatch = new CountDownLatch(1);
-        final CountDownLatch cleanupLatch = new CountDownLatch(1);
+        final Long READ_TIMEOUT = 250l;
 
-        Thread server = new Thread(new Runnable() {
+        //start a simple http server
+        SimpleHttpServer server = new SimpleHttpServer() {
             @Override
-            public void run() {
-                ServerSocket serverSocket = null;
-                Socket socket = null;
-                BufferedWriter w = null;
-                try {
-                    //create a socket on port 54321
-                    serverSocket = ServerSocketFactory.getDefault()
-                            .createServerSocket(54321);
-                    startupLatch.countDown();
-                    //wait for a connection
-                    socket = serverSocket.accept();
-
-                    // Just send a simple success response.
-                    w = new BufferedWriter(new OutputStreamWriter(socket
-                            .getOutputStream()));
-                    w.write("HTTP/1.0 200 OK");
-                    w.flush();
-                    //now sleep for longer than the timeout
-                    Thread.sleep(500);
-                } catch (Throwable e) {
-                    TEST_LOG.logger.log(Level.SEVERE, "Exception in readTimeout test server", e);
-                } finally {
-                    //make sure we count down to free up the test if something went wrong
-                    startupLatch.countDown();
-                    //close resources
-                    IOUtils.closeQuietly(w);
-                    IOUtils.closeQuietly(socket);
-                    IOUtils.closeQuietly(serverSocket);
-                    cleanupLatch.countDown();
-                }
+            protected void serverAction(InputStream is, OutputStream os) throws Exception {
+                //sleep for longer than the read timeout
+                Thread.sleep(READ_TIMEOUT * 2);

```

Comment:
Yeah there is no need for the server to actually do anything, fixed the comment in 2557419